include/tvm/runtime/container.h

```diff
@@ -274,6 +276,137 @@ class ADT : public ObjectRef {
   TVM_DEFINE_OBJECT_REF_METHODS(ADT, ObjectRef, ADTObj);
 };
 
+/*! \brief An object representing string. It's POD type. */
+class StringObj : public Object {
+ public:
+  /*! \brief The length of the string object. */
+  uint32_t size;
+
+  /*! \brief The pointer to string data. */
+  const char* data;
+
+ private:
+  /*! \brief String object which is moved from std::string container. */
+  class FromStd;
+
+  friend class String;
+};
+
+/*! \brief reference to string objects. */
+class String : public ObjectRef {
+ public:
+  /*!
+   * \brief Construct a new String object
+   *
+   * \param other The moved/copied std::string object
+   *
+   * \note If user passes const reference, it will trigger copy. If it's rvalue,
+   * it will be moved into other.
+   */
+  inline explicit String(std::string other);
+
+  /*!
+   * \brief Compare is equal to other std::string
+   *
+   * \param other The other string
+   */
+  inline bool operator==(std::string other) const;
+
+  /*!
+   * \brief Compare is not equal to other std::string
+   *
+   * \param other The other string
+   */
+  inline bool operator!=(std::string other) const;
+
+  /*!
+   * \brief Compare is equal to other char string
+   *
+   * \param other The other char string
+   */
+  inline bool operator==(const char* other) const;
+
+  /*!
+   * \brief Compare is not equal to other char string
+   *
+   * \param other The other char string
+   */
+  inline bool operator!=(const char* other) const;
+
+  /*!
+   * \brief Returns a pointer to the char array in the string.
+   *
+   * \return const char*
```

Comment:
add compare and substr as in https://en.cppreference.com/w/cpp/string/basic_string

Reply:
Added `compare`. Currently adding `substr` needs to copy `StringObj` and set char array pointer accordingly, this is expensive. Otherwise we can add string view like capability to `String`.